portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,53 +29,83 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.FetchOptions;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     
-    List<Post> posts = new ArrayList<>();
+    List<Post> posts = entityToList(results, request);
+    
+    response.setContentType("application/json;");
+    response.getWriter().println(convertToJson(posts));
+  }
 
-    for (Entity entity : results.asIterable()) {
-      long id = entity.getKey().getId();
-      long timestamp = (long) entity.getProperty("timestamp");
-      String name = (String) entity.getProperty("name");
-      String email = (String) entity.getProperty("email");
-      String comment = (String) entity.getProperty("comment");
-      
-      Post post = new Post(id, name, email, comment, timestamp);
-      posts.add(post);
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Entity commentEntity = createEntity(request);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    response.sendRedirect("/contact.html");
+  }
+
+  private int getCommentNum(HttpServletRequest request) {
+    String numChoiceString = request.getParameter("quantity");
+   
+    // Convert the input to an int.
+    int commentNum;
+    try {
+      commentNum = Integer.parseInt(numChoiceString);
+    } catch (NumberFormatException e) {
+      return 10;
```

Comment:
Why 10?

Reply:
10 is the default number that I chose for the amount of comments shown. I just changed this so ten is a final variable instead of hardcoding the number directly into the function.